pkg/splunk/enterprise/util.go

```diff
@@ -2272,3 +2272,50 @@ func getApplicablePodNameForK8Probes(cr splcommon.MetaObject, ordinalIdx int32)
 	}
 	return fmt.Sprintf("splunk-%s-%s-%d", cr.GetName(), podType, ordinalIdx)
 }
+
+// getClusterManagerCurrentImage gets the image of the pods of the clusterManager before any upgrade takes place,
+// returns the image, and error if something goes wrong
+func getCurrentImage(ctx context.Context, c splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (string, error) {
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetSplunkStatefulsetName(instanceType, cr.GetName()),
+	}
+	statefulSet := &appsv1.StatefulSet{}
+	err := c.Get(ctx, namespacedName, statefulSet)
+	if err != nil {
+		return "", err
+	}
+
+	image := statefulSet.Spec.Template.Spec.Containers[0].Image
```

Comment:
Can we make sure that the slice `statefulSet.Spec.Template.Spec.Containers` is not empty?